rotkehlchen/chain/evm/decoding/xdai_bridge/decoder.py

```diff
@@ -0,0 +1,117 @@
+from rotkehlchen.chain.evm.decoding.interfaces import DecoderInterface
+from rotkehlchen.chain.evm.decoding.structures import (
+    DEFAULT_DECODING_OUTPUT,
+    DecoderContext,
+    DecodingOutput,
+)
+from rotkehlchen.chain.evm.decoding.types import CounterpartyDetails, EventCategory
+from rotkehlchen.chain.evm.decoding.utils import bridge_match_transfer, bridge_prepare_data
+from rotkehlchen.types import ChainID, ChecksumEvmAddress, DecoderEventMappingType
+from rotkehlchen.utils.misc import hex_or_bytes_to_address, hex_or_bytes_to_int
+
+if TYPE_CHECKING:
+    from rotkehlchen.assets.asset import Asset
+    from rotkehlchen.chain.evm.decoding.base import BaseDecoderTools
+    from rotkehlchen.chain.evm.node_inquirer import EvmNodeInquirer
+    from rotkehlchen.user_messages import MessagesAggregator
+
+
+class XdaiBridgeCommonDecoder(DecoderInterface, metaclass=abc.ABCMeta):
+
+    def __init__(
+            self,
+            evm_inquirer: 'EvmNodeInquirer',
+            base_tools: 'BaseDecoderTools',
+            msg_aggregator: 'MessagesAggregator',
+            deposit_event: bytes,
+            withdrawal_event: Optional[bytes],  # withdrawal is currently unsupported on gnosis
+            bridge_address: ChecksumEvmAddress,
+            bridged_asset: 'Asset',
+            source_chain: ChainID,
+            target_chain: ChainID,
+    ) -> None:
+        super().__init__(
+            evm_inquirer=evm_inquirer,
+            base_tools=base_tools,
+            msg_aggregator=msg_aggregator,
+        )
+        self.bridge_address = bridge_address
+        self.bridged_asset = bridged_asset
+        self.deposit_event = deposit_event
+        self.withdrawal_event = withdrawal_event
+        self.source_chain = source_chain
+        self.target_chain = target_chain
+
+    def _decode_bridged_asset(self, context: DecoderContext) -> DecodingOutput:
+        """
+        Decodes a bridging event for `bridged_asset` specified.
+        Either a deposit or a withdrawal.
+        """
+        if context.tx_log.topics[0] not in {self.deposit_event, self.withdrawal_event}:
+            return DEFAULT_DECODING_OUTPUT
+
+        if context.tx_log.topics[0] == self.deposit_event:
+            from_address = context.transaction.from_address
+            to_address = self.bridge_address
+        else:
+            from_address = self.bridge_address
+            to_address = hex_or_bytes_to_address(context.tx_log.data[0:32])
+        amount = asset_normalized_value(
+            amount=hex_or_bytes_to_int(context.tx_log.data[32:64]),
+            asset=self.bridged_asset.resolve_to_crypto_asset(),
+        )
+        expected_event_type, new_event_type, from_chain, to_chain, expected_location_label = bridge_prepare_data(  # noqa: E501
+            tx_log=context.tx_log,
+            deposit_events=(self.deposit_event,),
+            main_chain=self.source_chain,
+            l2_chain=self.target_chain,
```

Comment:
Can you maybe adjust to what lefteris suggested here?

https://github.com/rotki/rotki/pull/6761#discussion_r1363738777